lib/processor.ts

```diff
@@ -1,44 +1,33 @@
 /**
  * @author Toru Nagashima <https://github.com/mysticatea>
  */
-'use strict'
+import type { Linter } from 'eslint'
 
-/**
- * @typedef {import('eslint').Linter.LintMessage} LintMessage
- */
-/**
- * @typedef {object} GroupState
- * @property {Set<string>} GroupState.disableAllKeys
- * @property {Map<string, string[]>} GroupState.disableRuleKeys
- */
+type LintMessage = Linter.LintMessage
+
+interface GroupState {
+  disableAllKeys: Set<string>
+  disableRuleKeys: Map<string, string[]>
+}
 
-module.exports = {
-  /** @param {string} code */
-  preprocess(code) {
+export default {
+  preprocess(code: string) {
     return [code]
   },
 
-  /**
-   * @param {LintMessage[][]} messages
-   * @returns {LintMessage[]}
-   */
-  postprocess(messages) {
+  postprocess(messages: LintMessage[][]) {
     const state = {
-      /** @type {GroupState} */
       block: {
-        disableAllKeys: new Set(),
-        disableRuleKeys: new Map()
-      },
-      /** @type {GroupState} */
+        disableAllKeys: new Set<string>(),
+        disableRuleKeys: new Map<string, string[]>()
+      } as GroupState,
```

Comment:
`as` is less safe than `satisfies`, so let's better use that one:

```suggestion
      } satisfies GroupState,
```